### DISCUSSION TITLE
How to restart TimeBase manually

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#256] [Q&A]
Recently, we came across TimeBase bug due to which my cursors were held open for long time.
So we need to restart TB manually. But I couldn't find any way out to restart? Please suggest how to do same.
Does tdbserver.sh script has some parameters/options as well

Previously,
when TimeBase got hanged due to "modify" column in a stream containing huge data, we tried killing all Process ids and then run tdbserver.sh script. But in this way we lose multiple spaces data in multiple streams.

**Comment by alex-karpovich** [comment]
you can send SIGTERM for process:
```
kill <pid>
```
or send shutdown signal via HTTP request: 
```
curl http://localhost:8011/shutdown...
```

**Comment by ankitsoin** [comment]
HI @alex-karpovich ,
While upgrading version of TimeBase instance, we always lose data/streams.
Can you please let us know the best recommended way to upgrade.

**Reply by ankitsoin** [reply]
@alex-karpovich , can you please suggest a best way.

**Reply by alex-karpovich** [reply]
Greetings, @ankitsoin 

I would suggest that you do flush() when closing loaders.

Also I mentioned before:

> you can send SIGTERM for process:
> 
> ```
> kill <pid> (for linux)
> ctrl-C (for windows)
> ```
> 
> or send shutdown signal via HTTP request:
> 
> ```
> curl http://localhost:8011/shutdown...
> ```

**Reply by ankitsoin** [reply]
Thank you @alex-karpovich will try doing same.
